lib/client.js

```diff
@@ -355,20 +356,36 @@ extend(Raven.prototype, {
   },
 
   captureException: function captureException(err, kwargs, cb) {
-    if (!(err instanceof Error)) {
-      // This handles when someone does:
-      //   throw "something awesome";
-      // We synthesize an Error here so we can extract a (rough) stack trace.
-      err = new Error(err);
-    }
-
     if (!cb && typeof kwargs === 'function') {
       cb = kwargs;
       kwargs = {};
     } else {
       kwargs = kwargs || {};
     }
 
+    if (!(err instanceof Error)) {
+      if (utils.isPlainObject(err)) {
+        // This will allow us to group events based on top-level keys
+        // which is much better than creating new group when any key/value change
+        var hash = md5(Object.keys(err).sort());
+        var message = 'Sentry: non-Error exception captured [' + hash + ']';
```

Comment:
I don't know that we need to prefix `Sentry:` here.

What about:

```
Non-Error exception captured with keys foo, bar, ...
```

^Something like that, only showing the first 2 or 3 keys.

Reply:
Updated